File: leetcode0150/solution.py
class Solution:
    def evalRPN(self, tokens: List[str]) -> int:
        stack = []

        current = 0

        operators = ["+", "-", "*", "/"]

        while current < len(tokens):
            if tokens[current] not in operators:
                stack.append(int(tokens[current]))
            else:
                v2 = stack.pop()
                v1 = stack.pop()
                match tokens[current]:
                    case "+":
                        stack.append(v1 + v2)
                    case "-":
                        stack.append(v1 - v2)
                    case "*":
                        stack.append(v1 * v2)
                    case "/":
                        stack.append(int(v1 / v2))
                    case _:
                        pass
            current += 1


        return stack[0]

# class Solution:
#     def evalRPN(self, tokens: List[str]) -> int:
#         stack = []

#         current = 0

#         while current < len(tokens):
#             if self.is_a_number(tokens[current]):
#                 stack.append(int(tokens[current]))
#             else:
#                 v2 = stack.pop()
#                 v1 = stack.pop()
#                 match tokens[current]:
#                     case "+":
#                         stack.append(v1 + v2)
#                     case "-":
#                         stack.append(v1 - v2)
#                     case "*":
#                         stack.append(v1 * v2)
#                     case "/":
#                         stack.append(int(v1 / v2))
#                     case _:
#                         pass
#             current += 1


#         return stack[0]

#     def is_a_number(self, x):
#         try:
#             int(x)
#             return True
#         except ValueError:
#             return False

File: leetcode0150/solution.rs
impl Solution {
    pub fn eval_rpn(tokens: Vec<String>) -> i32 {
        let mut stack: Vec<i32> = Vec::new();

        let mut current: usize = 0;

        while current < tokens.len() {

            if let Ok(val) = tokens[current].parse::<i32>() {
                stack.push(val);
            } else {
                let v2: i32 = stack.pop().unwrap();
                let v1: i32 = stack.pop().unwrap();
                match &*tokens[current] {
                    "+" => stack.push(v1 + v2),
                    "-" => stack.push(v1 - v2),
                    "*" => stack.push(v1 * v2),
                    "/" => stack.push(v1 / v2),
                    _ => {}
                }
            }

            current += 1;
        }
        
        *stack.last().unwrap()
    }
}